src/ActionModal/utils/ActionManager.js

```diff
@@ -149,20 +149,24 @@ function convertCurrencyData(amounts, network) {
 
 // limitation of the Ledger Nano S, so we pick the top 5 rewards and inform the user.
 export function getTop5RewardsValidators(rewards) {
-  const mostClaimableToken = rewards.reduce((topRewardToken, reward) => {
-    if (Number(reward.amount) > Number(topRewardToken.amount)) {
-      return reward
+  const rewardsPerValidatorObject = rewards.reduce((all, reward) => {
+    return {
+      ...all,
+      [reward.validator.operatorAddress]:
+        Number(reward.amount) +
```

Comment:
this sums up all rewards. but is ok I guess

Reply:
Yes. I also consider this since eJPY for example has a very bulky reward and is not so valuable. But I think it is OK for the time being (we don't really need that much precision for now)